sbin/firehol

```diff
@@ -25,31 +25,28 @@
 #       See the file COPYING for details.
 #
 
-VERSION='$Id$'
-PROGRAM_FILE="${0}"
-PROGRAM_DIR="${0%/*}"
-if [ "$PROGRAM_DIR" = "$0" ]; then PROGRAM_DIR="."; fi
+PROGRAM_FILE="$(/bin/readlink $0)"
+PROGRAM_FILE="${PROGRAM_FILE:-$0}"
```

Comment:
Actually, busybox readlink has no -f option... so I think the logic is needed to handle small platforms like my NAS and router.

Annoying, too is that it is not in `/bin` but `/usr/bin/` on two of my systems.

What seems better? Just assume it will be on the path, or contort the command to something like:

~~~~
PROGRAM_FILE="$(test -x /bin/readlink && /bin/readlink "$0" || /usr/bin/readlink "$0")"
~~~~